App.tsx

```diff
@@ -0,0 +1,168 @@
+import React from 'react';
+import Routes from './Routes';
+import _ from 'lodash';
+import {
+  View,
+  Text,
+  SafeAreaView
+} from 'react-native';
+import { fetchGutkas, saveGutkas, fetchSettings, findCurrentGutka, getGutkaItems } from './functions';
+import { storedGutka, itemsObj } from './Config/types';
+
+
+interface IProps { };
+interface IState {
+  gutkas: storedGutka[],
+  currentName: string,
+  currentItems: itemsObj[],
+  isDataReady: boolean,
+  isEditMode: boolean,
+  currShabadID: number,
+
+  gurmukhiSize: number,
+  translSize: number,
+  translitSize: number,
+  displayEngTransl: boolean,
+  displayPunTrasl: boolean,
+  displayTranslit: boolean,
+}
+class App extends React.Component<IProps, IState> {
+  constructor(props: IProps) {
+    super(props);
+    this.state = {
+      // init values
+      gutkas: [],
+      currentName: "",
+      currentItems: [],
+      isDataReady: false,
+      isEditMode: false,
+      currShabadID: 0,
+
+      //display settings
+      gurmukhiSize: 12,
+      translSize: 12,
+      translitSize: 12,
+      displayEngTransl: true,
+      displayPunTrasl: true,
+      displayTranslit: true,
+    }
+  }
+  async componentDidMount() {
+    const gutkasFetched = await fetchGutkas(this.state.currentName);
+    const { $isDataReady, $stored, $currentName, $currentItems } = gutkasFetched;
+    this.setState({ isDataReady: $isDataReady, gutkas: $stored, currentName: $currentName, currentItems: $currentItems });
+
+    const settingsFetched = await fetchSettings();
+    const { $displayEngTransl, $displayPunTrasl, $displayTranslit, $gurmukhiSize, $translSize, $translitSize } = settingsFetched;
+    this.setState({ displayEngTransl: $displayEngTransl, displayPunTrasl: $displayPunTrasl, displayTranslit: $displayTranslit, gurmukhiSize: $gurmukhiSize, translSize: $translSize, translitSize: $translitSize });
+  }
+
+  toggleEditMode = () => { this.setState((prevState) => ({ isEditMode: !prevState.isEditMode })); }
+
+  updateCurrentGutka = (newGutka: string) => {
+    const gutka = findCurrentGutka(this.state.gutkas, this.state.currentName);
+    this.setState({ currentName: newGutka, currentItems: getGutkaItems(gutka) });
+  }
+  updateCurrShabadID = (newID: number) => this.setState({ currShabadID: newID });
+
+  updateFontSize = (element: string, size: number) => {
+    const fontSetting = `${element}Size`;
+    this.setState(prevState => ({
+      ...prevState,
+      [fontSetting]: size,
+    }));
+  }
+  // removeFromGutka = (id: any) => {
+  //   let arr = this.state.currentItems;
+  //   const allGutkas = this.state.gutkas;
+  //   arr.splice(id, 1);
```

Comment:
Instead of splicing twice, once for `arr` and allGutkas, just splice allGutkas and do`_.values()` of the items to reset `currentItems`. Current implementation of this will not work as `arr` is just the values and setting it will mess w `allGutkas`